modules/tinymce/src/themes/silver/main/ts/ui/core/color/ColorSwatch.ts

```diff
@@ -187,19 +187,22 @@ const registerTextColorMenuItem = (editor: Editor, name: string, format: string,
 };
 
 const colorPickerDialog = (editor: Editor) => (callback, value: string) => {
-  const getOnSubmit = (callback) => (api) => {
+  let isValid = false;
+
+  const getOnSubmit = (callback) => (api: Dialog.DialogInstanceApi<ColorSwatchDialogData>) => {
     const data = api.getData();
-    callback(Optional.from(data.colorpicker));
-    api.close();
+    const hex = data.colorpicker;
+    if (isValid) {
+      callback(Optional.from(hex));
+      api.close();
+    } else {
+      editor.windowManager.alert(editor.translate([ 'Invalid hex color code: {0}', hex ]));
```

Comment:
Don't forget to make a PR against our lang pack repo as well to add this new string so that we know to get it translated.

- [ ] Add lang pack PR